src/printer.js

```diff
@@ -2080,8 +2080,20 @@ function genericPrintNoParens(path, options, print, args) {
     case "TSArrayType":
       return concat([path.call(print, "elementType"), "[]"]);
     case "TSPropertySignature":
+      var computed = !namedTypes.Identifier.check(n.name);
```

Comment:
Note that this change caused the test
`tests/typescript/conformance/types/union/__snapshots__/jsfmt.spec.js.snap` to add square brackets around the numbers (which is correct, but not required).